lib/brainstem/query_strategies/filter_and_search.rb

```diff
@@ -3,14 +3,15 @@ module QueryStrategies
     class FilterAndSearch < BaseStrategy
       def execute(scope)
         scope, ordered_search_ids = run_search(scope, filter_includes.map(&:name))
-        scope = @options[:primary_presenter].apply_filters_to_scope(scope, @options[:params], @options)
+        scope = primary_presenter.apply_filters_to_scope(scope, @options[:params], @options)
 
         if ordering?
           count_scope = scope
-          scope = paginate(scope)
-          scope = @options[:primary_presenter].apply_ordering_to_scope(scope, @options[:params])
-          primary_models = evaluate_scope(scope)
-          count = evaluate_count(count_scope)
+          scope = primary_presenter.apply_ordering_to_scope(scope, @options[:params])
+          limit, offset = calculate_limit_and_offset
+          ids = paginator.get_ids(limit: limit, offset: offset, scope: scope)
+          count = paginator.get_count(count_scope)
+          primary_models = Brainstem::QueryStrategies::DataMapper.new.get_models(ids: ids, scope: scope)
```

Comment:
We could extract a `data_mapper` function here and in `filter_or_search`. We could put it in the `base_strategy` class.